lib_standard_app/main.c

```diff
@@ -45,40 +38,35 @@ WEAK __attribute__((section(".boot"))) int main() {
 
     os_boot();
 
-    for (;;) {
-        // Initialize the UX system
-        UX_INIT();
+    // Initialize the UX system
+    UX_INIT();
 
-        BEGIN_TRY {
-            TRY {
-                io_seproxyhal_init();
+    BEGIN_TRY {
+        TRY {
+            io_seproxyhal_init();
 
 #ifdef HAVE_BLE
-                G_io_app.plane_mode = os_setting_get(OS_SETTING_PLANEMODE, NULL, 0);
+            G_io_app.plane_mode = os_setting_get(OS_SETTING_PLANEMODE, NULL, 0);
 #endif // HAVE_BLE
-                USB_power(0);
-                USB_power(1);
+            USB_power(0);
+            USB_power(1);
 
 #ifdef HAVE_BLE
-                BLE_power(0, NULL);
-                BLE_power(1, NULL);
+            BLE_power(0, NULL);
+            BLE_power(1, NULL);
 #endif // HAVE_BLE
-                app_main();
-            }
-            CATCH(EXCEPTION_IO_RESET) {
-                CLOSE_TRY;
-                continue;
-            }
-            CATCH_ALL {
-                CLOSE_TRY;
-                break;
-            }
-            FINALLY {
-            }
+
+            app_main();
+        }
+        CATCH_OTHER(e) {
```

Comment:
What is the difference between `CATCH_OTHER` and `CATCH_ALL` ?

Reply:
Discussed in direct